x-pack/plugins/ml/common/util/job_utils.ts

```diff
@@ -380,27 +382,70 @@ export function basicJobValidation(
         valid = false;
       }
     }
-
+    let categorizerDetectorMissingPartitionField = false;
     if (job.analysis_config.detectors.length === 0) {
       messages.push({ id: 'detectors_empty' });
       valid = false;
     } else {
       let v = true;
+
       each(job.analysis_config.detectors, (d) => {
         if (isEmpty(d.function)) {
           v = false;
         }
+        // if detector has an ml category, check if the partition_field is missing
+        const needToHavePartitionFieldName =
+          job.analysis_config.per_partition_categorization?.enabled === true &&
+          (d.by_field_name === MLCATEGORY || d.over_field_name === MLCATEGORY);
+
+        if (needToHavePartitionFieldName && d.partition_field_name === undefined) {
+          categorizerDetectorMissingPartitionField = true;
+        }
       });
       if (v) {
         messages.push({ id: 'detectors_function_not_empty' });
       } else {
         messages.push({ id: 'detectors_function_empty' });
         valid = false;
       }
+      if (categorizerDetectorMissingPartitionField) {
+        messages.push({ id: 'categorizer_detector_missing_per_partition_field' });
+        valid = false;
+      }
     }
 
-    // check for duplicate detectors
     if (job.analysis_config.detectors.length >= 2) {
+      // check if the detectors with mlcategory might have different per_partition_field values
+      // if per_partition_categorization is enabled
+      if (job.analysis_config.per_partition_categorization !== undefined) {
+        if (
+          job.analysis_config.per_partition_categorization.enabled ||
+          (job.analysis_config.per_partition_categorization.stop_on_warn &&
+            Array.isArray(job.analysis_config.detectors) &&
+            job.analysis_config.detectors.length >= 2)
+        ) {
+          const categorizationDetectors = job.analysis_config.detectors.filter(
+            (d) =>
+              d.by_field_name === MLCATEGORY ||
+              d.over_field_name === MLCATEGORY ||
+              d.partition_field_name === MLCATEGORY
+          );
+          const uniqPartitions = uniq(
+            categorizationDetectors
+              .map((d) => d.partition_field_name)
+              .filter((name) => name !== undefined)
+          );
```

Comment:
We should avoid using lodash functions in case ES standard provides something similar, please try to replace it with `Set` instead. 
```suggestion
          const uniqPartitions =[...new Set(
            categorizationDetectors
              .map((d) => d.partition_field_name)
              .filter((name) => name !== undefined)
          )];
```

Reply:
Updated in https://github.com/elastic/kibana/pull/75061/commits/38edcdd53460c2fab6fd1d36089c92ad8be8c78f